linode/manager
# refactor: [M3-7687] - Linodes Restricted User Experience 2/2

## Description 📝
Overall improvement to the Linode restricted experience. This PR disables lot's of buttons and fields that a user doesn't have access to.

## Changes  🔄
- Removed the banners from beneath each tab on the Linodes detail page since we will display an overall banner at the top of the page.
- If you don't have `add_linodes` grant, we disable create buttons. If you do have the grant, you will be able to initiate the creation of Linodes, as well as manage those you have personally established. However, restrictions may still apply for modifying of Linodes configured by the administrator.

## Target release date 🗓️
03/04/2024

## Preview 📷
### BEFORE
| | |
| ------- | ------- |
| ![Screenshot 2024-02-26 at 5 01 59 PM](https://github.com/linode/manager/assets/125309814/6a62d9e2-d5cc-49dc-8ea0-58ab16a2c38d) | ![Screenshot 2024-02-26 at 5 02 24 PM](https://github.com/linode/manager/assets/125309814/2df2287a-713a-4471-97b4-18aa98bcf28f) |
| ![Screenshot 2024-02-26 at 5 02 40 PM](https://github.com/linode/manager/assets/125309814/cd2fe900-8fae-44fe-b266-c65c86189647) | ![Screenshot 2024-02-26 at 5 02 07 PM](https://github.com/linode/manager/assets/125309814/c8f3ce3f-b244-4eda-924d-86bd26fabd32) |

### AFTER
| | |
| ------- | ------- |
|![Screenshot 2024-02-26 at 5 09 06 PM](https://github.com/linode/manager/assets/125309814/a3e6d2c5-c3f3-4b84-8237-7f9235461213) | ![Screenshot 2024-02-26 at 5 08 27 PM](https://github.com/linode/manager/assets/125309814/a988953f-4032-4d95-9dc0-66f9d676527f) |
| ![Screenshot 2024-02-26 at 5 14 58 PM](https://github.com/linode/manager/assets/125309814/87651147-d5d6-433e-af4f-ea1c1a5c28cb) | ![Screenshot 2024-02-26 at 5 15 33 PM](https://github.com/linode/manager/assets/125309814/bdc02d04-1aa4-466c-a42f-a7dc14a1a223) |
|![Screenshot 2024-02-26 at 5 20 57 PM](https://github.com/linode/manager/assets/125309814/351cccf6-67ed-4b64-8d56-3006adb3b6a8)| |


## How to test 🧪

### Prerequisites
(How to setup test environment)
- Log into two accounts side by side:
   - An unrestricted admin user account: full access
   - A restricted user account (use Incognito for this)
       - Start with `Read Only` for everything

### Reproduction steps
(How to reproduce the issue, if applicable)
- Create a linode from an unrestricted admin account and give your restricted user view only access.
- Observe that before changes, a user can attempt to edit/modify linodes that they don't have access to edit/modify and they will get denied by the API. This is what we're looking to prevent.

### Verification steps
(How to verify changes)
- Verify that restricted users can access only what they have permissions for.
- Verify that navigating via URL still display correct banners

## As an Author I have considered 🤔

*Check all that apply*
